crates/hir-ty/src/lib.rs

```diff
@@ -719,3 +721,16 @@ where
     value.visit_with(&mut collector, DebruijnIndex::INNERMOST);
     collector.placeholders.into_iter().collect()
 }
+
+pub fn known_const_to_string(konst: &Const, db: &dyn HirDatabase) -> Option<String> {
```

Comment:
I'm not sure where it would be better to place this function, nor how to name it. but it seems that it should be separated from the [old](https://github.com/rust-lang/rust-analyzer/blob/master/crates/hir-def/src/hir/type_ref.rs#L396-L408) `display` function, as it treats unknown consts in a special way and "unfolds" complex expressions.